powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/database/model/entity/OperationTemplateEntity.java

```diff
@@ -0,0 +1,236 @@
+    }
+
+    /**
+     * Set template ID.
+     * @param id Template ID.
+     */
+    public void setId(Long id) {
+        this.id = id;
+    }
+
+    /**
+     * Get template name.
+     * @return Template name.
+     */
+    public String getTemplateName() {
+        return templateName;
+    }
+
+    /**
+     * Set template name.
+     * @param templateName Template name.
+     */
+    public void setTemplateName(String templateName) {
+        this.templateName = templateName;
+    }
+
+    /**
+     * Get operation type.
+     * @return Operation type.
+     */
+    public String getOperationType() {
+        return operationType;
+    }
+
+    /**
+     * Set operation type.
+     * @param operationType Operation type.
+     */
+    public void setOperationType(String operationType) {
+        this.operationType = operationType;
+    }
+
+    /**
+     * Get data template. Template of the data used for signing. The value may contain `${xyz}` placeholders.
+     * @return Data template.
+     */
+    public String getDataTemplate() {
+        return dataTemplate;
+    }
+
+    /**
+     * Set data template. Template of the data used for signing. The value may contain `${xyz}` placeholders.
+     * @param dataTemplate Data template.
+     */
+    public void setDataTemplate(String dataTemplate) {
+        this.dataTemplate = dataTemplate;
+    }
+
+    /**
+     * Get signature type. Value representing which factors are allowed. One of: `1FA`, `2FA`, `2FA_NO_BIOMETRY`.
+     * @return Signature types.
+     */
+    public PowerAuthSignatureTypes[] getSignatureType() {
+        return signatureType;
+    }
+
+    /**
+     * Set signature type. Value representing which factors are allowed. One of: `1FA`, `2FA`, `2FA_NO_BIOMETRY`.
+     * @param signatureType Signature types.
+     */
+    public void setSignatureType(PowerAuthSignatureTypes[] signatureType) {
+        this.signatureType = signatureType;
+    }
+
+    /**
+     * Get maximum failure count.
+     * @return Maximum failure count.
+     */
+    public Long getMaxFailureCount() {
+        return maxFailureCount;
+    }
+
+    /**
+     * Set maximum failure count.
+     * @param maxFailureCount Maximum failure count.
+     */
+    public void setMaxFailureCount(Long maxFailureCount) {
+        this.maxFailureCount = maxFailureCount;
+    }
+
+    /**
+     * Get expiration in seconds (since "now").
+     * @return Expiration.
+     */
+    public Long getExpiration() {
+        return expiration;
+    }
+
+    /**
+     * Set expiration in seconds (since "now").
+     * @param expiration Expiration.
+     */
+    public void setExpiration(Long expiration) {
+        this.expiration = expiration;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (!(o instanceof OperationTemplateEntity)) return false;
+        OperationTemplateEntity that = (OperationTemplateEntity) o;
+        return Objects.equals(id, that.id)
```

Comment:
I believe this equals implementation is incorrect. The object is different before it is persisted into database and after persistence because the ID changes during persistence.

Reply:
Makes sense.